Generic ZigBee lock DTH with support for Yale touchscreen lever lock

tpmanley opened the pull request:
This is meant to be a generic ZigBee lock device type handler. The
initial testing was done with a Yale touchscreen level lock YRL220.

This version does not have support for configuring door lock codes.

workingmonk commented on devicetypes/smartthings/zigbee-lock.src/zigbee-lock.groovy:
```diff
@@ -0,0 +1,159 @@
+private Map parseReportAttributeMessage(String description) {
+    log.trace "parseReportAttributeMessage() --- description: $description"
+
+    Map descMap = (description - "read attr - ").split(",").inject([:]) { map, param ->
```
Instead of this closure here, you can use the zigbee.parseDescriptionAsMap which does the name thing

tpmanley replied:
Thanks for pointing that out - I'll update the code to use it instead

workingmonk commented on devicetypes/smartthings/zigbee-lock.src/zigbee-lock.groovy:
```diff
@@ -0,0 +1,159 @@
+
+// Lock capability commands
+def lock() {
+    def cmds = zigbee.zigbeeCommand("${CLUSTER_DOORLOCK}", "${DOORLOCK_CMD_LOCK_DOOR}", "{}")
```
this looks so clean :)

tpmanley commented:
This version of the DTH is complete and beginning certification. Once certification is complete I'll squash and merge unless there are any issues.

workingmonk commented:
@tpmanley can you comment out the YALE fingerprint and squash the commits?

workingmonk commented:
got a :+1:  from the certification team

tpmanley committed:
Generic ZigBee lock DTH w/ support for Kwikset models 910, 912, 914 and 916

This version does not have support for configuring door lock codes.

[DVCSMP-1030] ZigBee Kwikset Lock support

tpmanley commented:
Done. @workingmonk if the commit looks good then go ahead and merge

workingmonk commented:
Looks good :+1:
